pkg/storage/streams/store.go

```diff
@@ -626,3 +671,30 @@ func getEncryptedKeyAndNonce(m *pb.SegmentMeta) ([]byte, *eestream.Nonce) {
 
 	return m.EncryptedKey, &nonce
 }
+
+func getDecryptedStreamInfo(ctx context.Context, derivedKey *eestream.Key, item segments.Meta) (decryptedStreamInfo []byte, err error) {
+	streamMeta := pb.StreamMeta{}
+	err = proto.Unmarshal(item.Data, &streamMeta)
+	if err != nil {
+		return nil, err
+	}
+
+	cipher := eestream.Cipher(streamMeta.EncryptionType)
+	var nonce eestream.Nonce
+
+	encryptedKey, keyNonce := getEncryptedKeyAndNonce(streamMeta.LastSegmentMeta)
+
+	e, err := cipher.Decrypt(encryptedKey, derivedKey, keyNonce)
+	if err != nil {
+		return nil, err
+	}
+
+	var encKey eestream.Key
+	copy(encKey[:], e)
+
+	decryptedStreamInfo, err = cipher.Decrypt(streamMeta.EncryptedStreamInfo, &encKey, &nonce)
```

Comment:
Where does this `&nonce` get set? It is never initialized, so it is probably not the same nonce that the stream info was encrypted with